library/swift/Request.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+/// Represents an Envoy HTTP request. Use `RequestBuilder` to construct new instances.
+@objcMembers
+public final class Request: NSObject {
+  /// Method for the request.
+  public let method: RequestMethod
+  /// URL for the request.
+  public let url: URL
+  /// Headers to send with the request.
+  /// Multiple values for a given name are valid, and will be sent as comma-separated values.
+  public let headers: [String: [String]]
+  /// Trailers to send with the request.
+  /// Multiple values for a given name are valid, and will be sent as comma-separated values.
+  public let trailers: [String: [String]]
+  // Serialized data to send as the body of the request.
+  public let body: Data?
```

Comment:
I'd suggest adding `public let stream: Stream?` here also for body data. We can adopt one of two conventions - always send body data first, and then streamed data after, or enforce that only one of these is ever set during initialization. (The latter is probably safer and more common.)

We can also discuss and punt on this, but it's supported by Apple's own APIs and seems valuable to solve now (especially in line with @mattklein123's thoughts on the core API PR).

Reply:
Absolutely agree that streaming is something we need to support early on from the native interfaces. From our offline discussion, let's discuss outside of this PR's context, and add separately. Tracking in this issue: https://github.com/lyft/envoy-mobile/issues/247